### PR TITLE
Autorunner fixes2

### PR DESCRIPTION
**Pull request by myron**
This is the same PR as in https://github.com/Project-MONAI/MONAI/pull/5523

but removing changes to default ensemble method (AlgoEnsembleBestByFold), and keeping it as is for now, since it trigger errors with some unit tests.  the default ensemble method update can be done in a separate PR.

### Description

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by myron** [comment]
@wyli  here is the same PR as in https://github.com/Project-MONAI/MONAI/pull/5523
previously you had comment if the a new parameter "algo" should be placed at the very end or not. I think it should be in the front since it will a very commonly used parameter. Can you please comment if you still want "algo" at the end or not, and I can update it.  thank you

https://github.com/Project-MONAI/MONAI/pull/5523#discussion_r1029127063

**Comment by wyli** [comment]
sure, as long as it's not accidentally omitted...

**Comment by myron** [comment]
There is one integration test, I' unable to test in "test_integration_autorunner.py" -> test_autorunner_hpo(self) , test for HPO.  On my local machine it runs forever, and get stuck without any progress (no CPU or GPU usage).  Same behavior on old dev branch and on current (so it's unrelated to this PR).   So I've test everything except for HPO test there.
Perhaps we can merge it, as I don't think the changes in this PR should affect that test.

File: monai/apps/auto3dseg/auto_runner.py [hunk 1]
```diff
@@ -63,14 +64,13 @@ class AutoRunner:
...
             The configuration should contain datalist, dataroot, modality, multigpu, and class_names info.
-        analyze: on/off switch to run DataAnalyzer and generate a datastats report. If it is set to False,
-            The AutoRunner will attempt to skip datastats analysis and use cached results. If there is no such cache,
-            AutoRunner will report an error and stop.
-        algo_gen: on/off switch to run AlgoGen and generate templated BundleAlgos. If it is set to False,
-            The AutoRunner will attempt to skip the algorithm generation and stop if there is no cache to load.
-        train: on/off switch to run training and generate algorithm checkpoints. If it is set to False,
-            The AutoRunner will attempt to skip the training for all algorithms. If there is zero trained algorithm
-            but ``train`` is set to False, AutoRunner will stop.
+        algos: optionally specify a list of algorithms to use
```
[the remark this answers is not in the document]
Comment (myron) [reply monai/apps/auto3dseg/auto_runner.py]: Done.

**Review by mingxin-zheng — APPROVED** [review]

**Comment by myron** [comment]
I ran the integration test successfully too (test_integration_autorunner.py). The issue was that it does not run correctly on my local 8gpu machine, because a toy dateset there is too small.  It runs fine on 2gpu system. 

@wyli  please check if you have any comments. otherwise everything seems correct.

**Pull request merged**